### ISSUE TITLE
V2RC1 ACF Dynamic Data using Post Object field > Custom Field returns incorrect format

### ISSUE THREAD

**Issue by DigitalServicesLab** [soflyy/breakdance-bugs#1127]
When using the Post Object field for a dynamic data source, the option to manually use a ACF field key always returns the data direct from the database, not in the 'return format' defined by ACF. 

In the below example, the ACF field `people_profile_photo` is configured to return an image url. 
BD is returning the image ID (this can be seen if you use the dynamic data on a text field)

![image](https://github.com/user-attachments/assets/b3700c60-4fbc-489e-bdfa-219b12285eb7)

**Comment by DigitalServicesLab** [comment]
Having thought about this further, I guess this isn't a bug in as much as the custom field key chosen could be any field and BD doesn't know it's an ACF field - despite the fact its a sub field/child field of an ACF Post Object field.

This probably needs a checkbox to state 'this is an ACF' field and BD uses its ACF methods to return it. 
Or additional, use some contextual logic to understand what's being returned or what's needed... in this case the dynamic data is on an image element, so you'd expect it to know it needs an image url (or image atts array), but the return format is controlled by the PostField field Class which just returns a string... always. 

As an aside from that, there doesn't seem to be a way to return a literal image url string to an image element from a custom field either. I used the ACF `'acf/save_post'` action to save the actual image url in a separate meta field after ACF saved the image id but even if I use that as the field key, it still doesn't work, so what is the correct format required for a dynamic image data? 
When selecting dynamic data options, it says 'image url' but that obviously isn't really what it is returning or at least it wants it in a very specific format or needs to process it before it gets it. 
I tried:

- a image ID
- an image URL
- an image attributes array
- an full html image tag

none of this works. 


In a nutshell - Post Object fields have limited use with custom fields, unless some context awareness is included, or ACF field logic is included. Otherwise its usefulness extends only to text data.

**Comment by DigitalServicesLab** [comment]
Given that a Post Object field provides access to another Post (not the current global post), I think the best solution would be to provide access to the normal Dynamic Data field selection screen but in the context of the post of the Post Object. 

So select dynamic data > Post Object field
And then in the options for that have another button that opens the dynamic data screen again but now you are selecting the field for the post object... makes more sense than having a limited drop down of text fields or custom keys to select from.

**Comment by andyfo** [comment]
Just came here to say I was going to let a user assign a CPT to a page via ACF's Post Object, and I was hoping I could use the CPT's fields via the Post Object to display the content. Breakdance lets me select the ACF fields from the CPT, but won't actually show anything (either image or text).
